### PR TITLE
Rpaasv2/go client

### PR DESCRIPTION
**Pull request by arthurcgc**

### CODE REVIEW

File: pkg/rpaas/client/client.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+func New(hostAPI string) *RpaasClient {
+	client := &RpaasClient{httpClient: &http.Client{}, hostAPI: hostAPI}
+	return client
+}
+
+func NewTsuruClient(tsuruAPI, service, token string) (*RpaasClient, error) {
+	if service == "" || tsuruAPI == "" {
+		return nil, fmt.Errorf("service can't be nil")
+	}
```
Comment (nettoclaudio) [inline pkg/rpaas/client/client.go]:
```suggestion
	if tsuruAPI == "" || token == "" || service == "" {
		return nil, fmt.Errorf("cannot create client without either tsuru target, token or service")
	}
```

File: pkg/rpaas/client/client.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+type RpaasClient struct {
+	hostAPI    string
+	tsuruAPI   string
```
Comment (nettoclaudio) [inline pkg/rpaas/client/client.go]: I would rename to `tsuruTarget` instead...

File: pkg/rpaas/client/client.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+	token      string
```
Comment (nettoclaudio) [inline pkg/rpaas/client/client.go]: `tsuruToken`...

File: pkg/rpaas/client/client.go [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+	target     string
+	service    string
```
Comment (nettoclaudio) [inline pkg/rpaas/client/client.go]: and `tsuruService`.

File: pkg/rpaas/client/client.go [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetRowLine(true)
+	table.SetHeader([]string{prefix, "Description"})
+	for _, v := range dataSlice {
+		table.Append(v)
+	}
+
+	table.Render()
+}
```
Comment (nettoclaudio) [inline pkg/rpaas/client/client.go]: Drop this copy-and-pasted code...

File: pkg/rpaas/client/client.go [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+			return fmt.Errorf("Error while trying to read body: %v", err)
+		}
+		var fp interface{}
+		err = json.Unmarshal(body, &fp)
+		if err != nil {
+			return err
+		}
+		helperSlice := fp.([]interface{})
+		writeInfo(infoType, helperSlice)
+		return nil
```
Comment (nettoclaudio) [inline pkg/rpaas/client/client.go]: WFT???

File: pkg/rpaas/client/client.go [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+	}
+	return &RpaasClient{
+		httpClient: &http.Client{},
+		tsuruAPI:   tsuruAPI,
+		token:      token,
+		service:    service,
+	}, nil
+}
+
+func (c *RpaasClient) Info(ctx context.Context, instance string, infoType string) error {
```
Comment (nettoclaudio) [inline pkg/rpaas/client/client.go]: Does the info method return only an error?

File: go.sum [hunk 8]
```diff
@@ -637,6 +637,7 @@ gopkg.in/yaml.v2 v2.0.0-20170812160011-eb3733d160e7/go.mod h1:JAlM8MvJe8wmxCU4Bl
 gopkg.in/yaml.v2 v2.2.1/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.2.2 h1:ZCJp+EgiOT7lHqUV2J862kp8Qj64Jo6az82+3Td9dZw=
 gopkg.in/yaml.v2 v2.2.2/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
+gotest.tools v2.2.0+incompatible h1:VsBPFP1AI068pPrMxtb/S8Zkgf9xEmTLJjfM+P5UIEo=
```
Comment (nettoclaudio) [inline go.sum]: You shouldn't change the root go mod.

File: pkg/rpaas/client/client.go (hunk 5 above)
Comment (arthurcgc) [reply pkg/rpaas/client/client.go]: those are the methods that create the ASCII tables from the tsuru target responses...

**Pull request closed without merging**